src/lib/util.ts

```diff
@@ -0,0 +1,49 @@
+import { Component, ComponentProps } from 'preact';
+
+type WhenProps = ComponentProps<When> & {
+  value: boolean,
+  children?: (JSX.Element | (() => JSX.Element))[]
+};
+
+type WhenState = {
+  ready: boolean
+};
+
+export class When extends Component<WhenProps, WhenState> {
+  state: WhenState = {
+    ready: !!this.props.value
+  };
+
+  render({ value, children = [] }: WhenProps, { ready }: WhenState) {
+    let child = children[0];
+    if (value && !ready) this.setState({ ready: true });
```

Comment:
What's the benefit of `ready` here? Can't you just test `value`?

Reply:
would create an infinite loop if the component loaded was undefined